### PR TITLE
Feature/refset html

### PR DESCRIPTION
**Pull request by melferink**
Included RefSet's in HTML view (UCSC)

### CODE REVIEW

File: ExomeDepth/make_BEDdetail.py [hunk 1]
```diff
@@ -208,7 +217,7 @@ def slice_vcf(args, merge_dic):
                 event_dic[event][sampletype]["deldupratio"].append(deldupratio)
                 event_dic[event][sampletype]["totalcalls"].append(totalcalls)
                 event_dic[event][sampletype]["gender"].append(gender)  # Not used in output at the moment.
-
+                event_dic[event][sampletype]["refset"].append(refset)
     all_event_file.close()
```
Comment (FiniDG) [inline ExomeDepth/make_BEDdetail.py:221]: here I see a file.close() with an `= open()` earlier in your file. However it is best practise (python 3) to use `with open()`, like you also did in line 60 for `vcf_input_file`. python will automatically close the file at the right time.
might be an improvement to make this more uniform and more python3.

File: ExomeDepth/make_BEDdetail.py [hunk 2]
```diff
@@ -49,6 +50,12 @@ def slice_vcf(args, merge_dic):
     event_dic = {}
     children = 0
     parents = 0
+
+    # Random shuffle to truly select random samples in subset (function calculate_statistics)
+    # Random shuffle has not influence on general statistics
```
Comment (FiniDG) [inline ExomeDepth/make_BEDdetail.py:55]: typo, 
additionally, is line 55 a code comment? maybe exclude this information

File: ExomeDepth/make_BEDdetail.py [hunk 3]
```diff
@@ -49,6 +50,12 @@ def slice_vcf(args, merge_dic):
...
+    if not args.random_off:
+        random.shuffle(vcf_files)
```
Comment (FiniDG) [inline ExomeDepth/make_BEDdetail.py:57]: I am not sure about this one, but should it be something like
`vcf_files = random.shuffle(vcf_files)` or will vcf_files update automatically?

File: ExomeDepth/make_BEDdetail.py [hunk 4]
```diff
@@ -230,16 +239,22 @@ def calculate_statistics(data, status):
         'ratio_stdev_{0}'.format(status): "n/a",
         'bfs_{0}'.format(status): "n/a",
         'ratios_{0}'.format(status): "n/a",
+        'refsets_{0}'.format(status): "n/a"
```
Comment (FiniDG) [inline ExomeDepth/make_BEDdetail.py:242]: before it ended with a comma

File: ExomeDepth/make_BEDdetail.py (hunk 3 above)
Comment (melferink) [reply ExomeDepth/make_BEDdetail.py:57]: Nope:

`>>> import random
>>> testlist = ["test2", "test3", "test1"]
>>> random.shuffle(testlist)
>>> testlist
['test1', 'test3', 'test2']
>>> random.shuffle(testlist)
>>> testlist
['test2', 'test3', 'test1']
>>> testlist = random.shuffle(testlist)
>>> testlist
>>>`

random sort is within the list

File: ExomeDepth/make_BEDdetail.py (hunk 4 above)
Comment (melferink) [reply ExomeDepth/make_BEDdetail.py:242]: Do you refer to each item with list "stats", or to a different list?
The comma in this list is not needed for the last item.

File: ExomeDepth/make_BEDdetail.py (hunk 2 above)
Comment (melferink) [reply ExomeDepth/make_BEDdetail.py]: comments removed.

File: ExomeDepth/make_BEDdetail.py [hunk 5]
```diff
@@ -390,6 +403,13 @@ def make_bed_detail(args, event_dic, children, parents):
         '--correlqc', default=settings.correlation, type=float,
         help='Threshold for minimum allowed correlation score (default = 0.98)'
     )
+    parser.add_argument(
+        '--random_off', action='store_true',
+        help=(
+            'Do not apply randomisation in VCF-file order.'
+            ' This is usefull for validation purposes if same results are needed (default = random selection on'
```
Comment (lvanoijen) [inline ExomeDepth/make_BEDdetail.py:410]: Missing a ")" at the end of the sentence

**Review by lvanoijen — CHANGES_REQUESTED** [review]
Comment Fini not yet resolved. And small typo in a comment.

File: ExomeDepth/make_BEDdetail.py (hunk 1 above)
Comment (melferink) [reply ExomeDepth/make_BEDdetail.py]: Good point, changed.

File: ExomeDepth/make_BEDdetail.py (hunk 5 above)
Comment (melferink) [reply ExomeDepth/make_BEDdetail.py]: I even detected 2 of those :)

**Review by lvanoijen — APPROVED** [review]
Approved

**Pull request merged**